### PR TITLE
Document implications of agent-less mode

### PR DESCRIPTION
**Pull request by uric**

### CODE REVIEW

File: content/agents/overview.md [hunk 1]
```diff
@@ -54,6 +54,15 @@ Following are all the supported installation methods:
...
 * `provided` - An agent is assumed to already be installed on the host image. That agent will be configured and started via a script that will run on the host when it gets created. This method is only supported for specific IaaS plugins.
 
+{{% gsNote title="Implications of Not Installing an Agent" %}}
+In some cases, you cannot or prefer not to install an agent on Cloudify-managed VMs. This could be due to a security restriction or the fact that a VM is a pre-baked closed appliance that you can't access or modify. 
+In such cases, you would use `install_method: none` to instruct Cloudify not to install an agent on the created VM. 
+However, you should be aware of certain implications: 
+
+* You will not be able to use certain plugins, specifically ones that assume execution on the agent's VM, i.e. ones that are configured with `executor=central_deployment_agent`. This includes Docker, Chef and Puppet plugins among others. To work around this you will need to run bash or Python scripts using the [Fabric plugin](../../plugins/fabric/) (e.g. invoke the Puppet client from a script instead of using the Puppet plugin).
```
Comment (idanmo) [inline content/agents/overview.md]: Please use the directives for links, available here: https://github.com/cloudify-cosmo/docs.getcloudify.org/blob/3.3.1-build/content/cheatsheet.md

File: content/agents/overview.md [hunk 2]
```diff
@@ -54,6 +54,15 @@ Following are all the supported installation methods:
...
 
+{{% gsNote title="Implications of Not Installing an Agent" %}}
+In some cases, you cannot or prefer not to install an agent on Cloudify-managed VMs. This could be due to a security restriction or the fact that a VM is a pre-baked closed appliance that you can't access or modify. 
+In such cases, you would use `install_method: none` to instruct Cloudify not to install an agent on the created VM. 
+However, you should be aware of certain implications: 
+
+* You will not be able to use certain plugins, specifically ones that assume execution on the agent's VM, i.e. ones that are configured with `executor=central_deployment_agent`. This includes Docker, Chef and Puppet plugins among others. To work around this you will need to run bash or Python scripts using the [Fabric plugin](../../plugins/fabric/) (e.g. invoke the Puppet client from a script instead of using the Puppet plugin). 
+* You will not be able to install a [Diamond monitoring agent](http://diamond.readthedocs.org/) using the [Diamond plugin](../../plugins/diamond/) because this plugin requires an agent to run. Naturally you can always install your own monitoring agent using a cloud init / the Fabric plugin.
```
Comment (idanmo) [inline content/agents/overview.md]: Same comment

File: content/agents/overview.md (hunk 1 above)
Comment (uric) [inline content/agents/overview.md]: you mean relref? still not seeing how it's better TBH...

Comment (ChenRoth) [inline content/agents/overview.md]: relref is used for 2 complementary reasons:
1) if you don't use it and you point to a root relative url (not the case here, but explained anyway), the link loses its branch context and just points to a docs.getcloudify.org/ page
2) it provides a way to write root relative paths, so it doesn't depend on the location of the page you're referencing from (alas, it still depends on the referenced page location)

**Pull request merged**